language/compiler/src/lib.rs

```diff
@@ -46,66 +44,66 @@ pub struct Compiler<'a> {
     pub _non_exhaustive: (),
 }
 
-impl<'a> Compiler<'a> {
+impl Compiler {
     /// Compiles into a `CompiledProgram` where the bytecode hasn't been serialized.
-    pub fn into_compiled_program(mut self) -> Result<CompiledProgram> {
-        Ok(self.compile_impl()?.0)
+    pub fn into_compiled_program(mut self, code: &str) -> Result<CompiledProgram> {
+        Ok(self.compile_impl(code)?.0)
     }
 
     /// Compiles into a `CompiledProgram` and also returns the dependencies.
     pub fn into_compiled_program_and_deps(
         mut self,
+        code: &str,
     ) -> Result<(CompiledProgram, Vec<VerifiedModule>)> {
-        self.compile_impl()
+        self.compile_impl(code)
     }
 
     /// Compiles into a `CompiledScript`.
-    pub fn into_script(mut self) -> Result<CompiledScript> {
-        let compiled_program = self.compile_impl()?.0;
+    pub fn into_script(mut self, code: &str) -> Result<CompiledScript> {
+        let compiled_program = self.compile_impl(code)?.0;
         Ok(compiled_program.script)
     }
 
     /// Compiles the script into a serialized form.
-    pub fn into_script_blob(mut self) -> Result<Vec<u8>> {
-        let compiled_program = self.compile_impl()?.0;
+    pub fn into_script_blob(mut self, code: &str) -> Result<Vec<u8>> {
+        let compiled_program = self.compile_impl(code)?.0;
 
         let mut serialized_script = Vec::<u8>::new();
         compiled_program.script.serialize(&mut serialized_script)?;
         Ok(serialized_script)
     }
 
     /// Compiles the module.
-    pub fn into_compiled_module(mut self) -> Result<CompiledModule> {
-        Ok(self.compile_mod()?.0)
+    pub fn into_compiled_module(mut self, code: &str) -> Result<CompiledModule> {
+        Ok(self.compile_mod(code)?.0)
     }
 
     /// Compiles the module into a serialized form.
-    pub fn into_module_blob(mut self) -> Result<Vec<u8>> {
-        let compiled_module = self.compile_mod()?.0;
+    pub fn into_module_blob(mut self, code: &str) -> Result<Vec<u8>> {
+        let compiled_module = self.compile_mod(code)?.0;
 
         let mut serialized_module = Vec::<u8>::new();
         compiled_module.serialize(&mut serialized_module)?;
         Ok(serialized_module)
     }
 
     /// Compiles the code and arguments into a `Script` -- the bytecode is serialized.
-    pub fn into_program(mut self, args: Vec<TransactionArgument>) -> Result<Script> {
-        let compiled_program = self.compile_impl()?.0;
-
+    pub fn into_program(mut self, code: &str, args: Vec<TransactionArgument>) -> Result<Script> {
```

Comment:
Ins't the body of this just
```
Ok(Script::new(self.into_script_block(code)?, args))
```
?

Reply:
Yeah, the body is a bit silly. I was doing pure refactoring without trying to simplify in this PR, but this could definitely use a simplification pass.